Mathlib/CategoryTheory/Limits/IsConnected.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2024 Paul Reichert. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+import Mathlib.CategoryTheory.Limits.Types
+import Mathlib.CategoryTheory.IsConnected
+
+/-!
+# Colimits of connected index categories
+
+This file proves that a category $\mathsf{C}$ is connected if and only if `colim F` is a singleton,
+where `F.obj _ = PUnit` (in a `Type _` category).
+
+See `connected_iff_colimit_const_pUnit_iso_pUnit` for the proof of this characterization and
```

Comment:
Here, you should update the name of lemma.

Reply:
Thanks, I've updated the names in the docstring and also incorporated other new lemmas.